### ISSUE TITLE
[Config Support]: Reolink delay & (Reolink E1) camera specific config

### ISSUE THREAD

**Issue by albotroz** [blakeblackshear/frigate#9464]
### Describe the problem you are having

After some time of testing my cameras within Frigate, I noticed that my Reolink camera reaches a delay of 20(!) seconds after about one day.
This is noticeable in the Frigate Addon, HassOS Frigate Card and Hass-Android-App etc.
Everything else worked so well and now I'm back to playing with the configs just because of this.

This is what the config for the Reolink camera looked like.
HD stream (without sound - I don't want sound in records because of Law-Stuff in my country) records events. SD stream for livestream and I added a button within Frigate Card to switch between HD and SD livestream with sound in the HAOS dashboard/frigate card.

No problems at all. Until the delay after one day of working.

```
ffmpeg:
  hwaccel_args: preset-rpi-64-h264

rtmp:
  enabled: false

go2rtc:
  streams:
    front:
      - rtsp://NAME:PW@192.168.xxx.xxx:554/h264Preview_01_main
      - "ffmpeg:front#audio=opus"
    front_sub:
      - rtsp://NAME:PW@192.168.xxx.xxx:554/h264Preview_01_sub
      - "ffmpeg:front_sub#audio=opus"

#  webrtc:
#    candidates:
#      - 192.168.xxx.xx:8555
#      - stun:8555
```

```
cameras:
  front:
    ffmpeg:
      output_args:
        record: preset-record-generic
      inputs:
        - path: rtsp://127.0.0.1:8554/front # RTSP Stream High Res
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/front_sub # RTPS Stream Low Res
          input_args: preset-rtsp-restream
          roles:
            - detect
```

I then changed it as in the docs. https://docs.frigate.video/configuration/camera_specific/#reolink-cameras
So far I haven't noticed any delay.
**However, this means that the sound in the live stream sounds laggy and there are no more recordings for events.**
Yes, next time I will get the cameras that Frigate recommended in the docs.

**This is how I changed it:**

```
ffmpeg:
  hwaccel_args: preset-rpi-64-h264

rtmp:
  enabled: false

go2rtc:
  streams:
    front:
      - "ffmpeg:http://192.168.xxx.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=NAME&password=PW#video=copy#audio=copy#audio=opus"
    front_sub:
      - "ffmpeg:http://192.168.xxx.xxx/flv?port=1935&app=bcs&stream=channel0_sub.bcs&user=NAME&password=PW"

#  webrtc:
#    candidates:
#      - 192.168.xxx.xx:8555
#      - stun:8555
```

```
cameras:
  front:
    ffmpeg:
#      output_args:
#        record: preset-record-generic
      inputs:
        - path: rtsp://127.0.0.1:8554/front?video=copy&audio=acc # RTSP Stream High Res
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/front_sub?video=copy # RTPS Stream Low Res
          input_args: preset-rtsp-restream
          roles:
            - detect
```

How should I interpret the FFmpeg parameters?
Are the streams taken from the camera with video=copy and audio=copy without transcoding or how?
What if only audio=opus is set? Then it is also taken without transcoding and transcoded within Raspberry to opus?
I want to understand the purpose of the FFmpeg parameters and not just to solve my problems.
And if I adopt the setting of Frigate-Reolink-Doc, then it causes a CPU utilization of almost 100% and temperature rises to 90 degrees + when watching the livestream.

### Version

0.12.1

### Frigate config file

```yaml
.
```

### Relevant log output

```shell
.
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

**Comment by NickM-27** [comment]
what is the cpu usage when using the recommended config? This is one of the reasons that using rtsp with reolink is not recommended.

**Comment by albotroz** [comment]
> what is the cpu usage when using the recommended config? This is one of the reasons that using rtsp with reolink is not recommended.

If I open HomeAssistant Dashboard with my both Sub-Livestreams from Reolink and Non-Reolink Cams:
Recommended config: 80-100 CPU Usage (laptop partially freezes), after few minutes 100 degrees
My config config: max 40 CPU Usage, max 85 degrees

**Comment by NickM-27** [comment]
Ah I see, using `_sub` instead of `_ext`, that is fine as long as you are using http and not rtsp in the go2rtc section. Regardless, given that the delay occurs in each place it is viewed that means the delay is likely coming from the camera where the stream timestamps fall behind real time. I had an issue similar with a reolink e1 zoom back when I had it, updating to a beta firmware from reolink seemed to fix the issue

**Comment by albotroz** [comment]
Since watching the livestream was impossible due to the CPU load, I reverted to my config. So the one with rtsp. This has worked excellently so far, although it is not recommended according to Frigate Docs. Only Delay was a problem. Theoretically, I would have to restart HASS/Frigate every day to reset the delay. Recording was not noticeable with the Frigate Recommended Reolink config.
When I then reverted to my config, Frigate still did not make any recordings with the Reolink. Blueprints notifications also stopped coming.

Basically, it no longer works with Reolink and I now have to revert back to Reolink internal recording and Reolink's push notifications. Don't know what happened between Reolink and Frigate.

**Comment by NickM-27** [comment]
try this config

```
ffmpeg:
  hwaccel_args: preset-rpi-64-h264

cameras:
  front:
    ffmpeg:
      input_args: preset-http-reolink
      inputs:
        - path: http://192.168.xxx.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=NAME&password=PW
          roles:
            - record
        - path: http://192.168.xxx.xxx/flv?port=1935&app=bcs&stream=channel0_sub.bcs&user=NAME&password=PW
          roles:
            - detect
```

**Comment by NickM-27** [comment]
Looks fine to me on a quick glance

**Comment by NickM-27** [comment]
Feel free to create a new issue if something else comes up
